test/smtchecker.js

```diff
@@ -1,25 +1,107 @@
 const tape = require('tape');
+const semver = require('semver');
+const solc = require('../index.js');
 const smtchecker = require('../smtchecker.js');
+const smtsolver = require('../smtsolver.js');
 
+const preamble = 'pragma solidity >=0.0;\n// SPDX-License-Identifier: GPL-3.0\n';
+//
 tape('SMTChecker', function (t) {
+  // We use null for `solverFunction` and `solver` when calling `handleSMTQueries`
+  // because theses tests do not call a solver.
+
   t.test('smoke test with no axuiliaryInputRequested', function (st) {
     const input = {};
     const output = {};
-    st.equal(smtchecker.handleSMTQueries(input, output), null);
+    st.equal(smtchecker.handleSMTQueries(input, output, null, null), null);
     st.end();
   });
 
   t.test('smoke test with no smtlib2queries', function (st) {
     const input = {};
     const output = { auxiliaryInputRequested: {} };
-    st.equal(smtchecker.handleSMTQueries(input, output), null);
+    st.equal(smtchecker.handleSMTQueries(input, output, null, null), null);
     st.end();
   });
 
   t.test('smoke test with empty smtlib2queries', function (st) {
     const input = {};
     const output = { auxiliaryInputRequested: { smtlib2queries: { } } };
-    st.equal(smtchecker.handleSMTQueries(input, output), null);
+    st.equal(smtchecker.handleSMTQueries(input, output, null, null), null);
+    st.end();
+  });
+
+  t.test('smtCallback should return type function', (st) => {
+    const response = smtchecker.smtCallback(() => {});
+    st.equal(typeof response, 'function');
+    st.end();
+  });
+
+  t.test('smtCallback should error when passed parser fails', (st) => {
+    const callback = smtchecker.smtCallback((content) => { throw new Error(content); });
+    // eslint-disable-next-line node/no-callback-literal
+    const response = callback('expected-error-message');
```

Comment:
Is this complaining that the variable is called `callback` and that's reserved on node? Why not just call it something else and avoid the warning? `testCallback` or something.

Reply:
I copied this from https://github.com/ethereum/solc-js/pull/566

Reply:
Will change